File: apps/ios/FamilyEvents/Networking/ConsumerAPIPath.swift
import Foundation

enum ConsumerAPIPath: Equatable {
    case events
    case eventDetail(id: String)
    case favorites
    case profile

    var value: String {
        switch self {
        case .events:
            return "/api/v1/events"
        case .eventDetail(let id):
            return "/api/v1/events/\(id)"
        case .favorites:
            return "/api/v1/favorites"
        case .profile:
            return "/api/v1/profile"
        }
    }
}

File: apps/android/core/src/main/java/com/familyevents/core/ConsumerApiPath.kt
package com.familyevents.core

sealed interface ConsumerApiPath {
    val value: String

    data object Events : ConsumerApiPath {
        override val value = "/api/v1/events"
    }

    data object Favorites : ConsumerApiPath {
        override val value = "/api/v1/favorites"
    }

    data object Profile : ConsumerApiPath {
        override val value = "/api/v1/profile"
    }

    data class EventDetail(val id: EventId) : ConsumerApiPath {
        override val value = "/api/v1/events/${id.rawValue}"
    }
}
